WooCommerce/Classes/ViewRelated/CardPresentPayments/BuiltInCardReaderConnectionController.swift

```diff
@@ -366,6 +376,29 @@ private extension BuiltInCardReaderConnectionController {
             .store(in: &self.subscriptions)
         }
         stores.dispatch(softwareUpdateAction)
+
+
+        if featureFlagService.isFeatureFlagEnabled(.tapToPayEducation), let presenter = merchantEducationPresenter {
+            let onboardingAction = CardPresentPaymentAction.observeBuiltInCardReaderAcceptToS { [weak self] events in
+                guard let self else { return }
+
+                events
+                    .subscribe(on: DispatchQueue.main)
+                    .sink { [weak self] in
+                        guard let self else { return }
+
+                        isMerchantEducationInProgress.send(true)
+                        presenter.presentMerchantEducation { [weak self] in
+                            guard let self else { return }
+                            isMerchantEducationInProgress.send(false)
```

Comment:
Thanks. I began by changing `state` here and using associated values. This is what we've spoken about previously. There were a couple of issues and overcomplications with it:

- I couldn't programmatically guarantee that education only happens in specific states. In practice, it could be visible when connecting, updating, and connecting. But if it happens at times I wouldn't expect, I wouldn't be able to set the `state`. It could be a minor issue and not likely to happen. 
- Adding associated values to existing states meant that the education state needed to be set at any time one of those states was set. `state = .connectToReader(education: false)`,  `state = .updating(progress:0, education: state.educationState)`. It felt a bit too intrusive.

> That would be the way to keep returnSuccess as a synchronous action, as we'd know whether education was in progress or not before we called it.

True. I would still need to think about how to keep it synchronous. Likely not calling `returnSuccess` if state contains `education` being in progress. But then calling `returnSuccess` when merchant presentation is dismissed and we have specific `state`. Could be a similar concern to the previous point where we would affect multiple code paths in the controller.

Overall, setting a separate state var simplified the approach but could be considered inconsistent since it didn't use the existing mechanism of the controller. I'll think one more time if existing state setup could be used without introducing too many changes or risk.